### DISCUSSION TITLE
v1.1.5 "Cherenkov"

### DISCUSSION THREAD

**Discussion by numeredev** [numere-org/NumeRe#102] [Announcements]
![release](https://github.com/numere-org/NumeRe/assets/87755490/81f67d66-5df6-4b2a-be83-7e52e9726789)

# Highlighted changes in this version

- It's now possible to define an enumeration using the syntax `declare enum -> {VAL1, VAL2, ...}`
- The documentation browser can now be opened again using `help window` and similar. The necessary time for opening up with this method was greatly reduced.
- Empty cells are now preserved, if one converts string columns to logical columns.
- Multiple head rows in tables and data files are now much better supported for XLS, XLSX, ODS and CSV
- Copying and moving (using `copy` and `move`) of mixed-value tables is now possible again.
- It's now possible to deactivate the table auto-save by setting the interval to zero
- Downloading of zip files now works as expected.
- The calculation of needed memory per table and cluster does now reflect better the actually needed memory.
- `TAB().num` now also considers strings as valid values for the `num()` operation and empty columns are now correctly calculated as empty if used with `TAB().cnt.cols`
- Improved performance when copying a large table out of the table viewer.
- Procedures execute now always the latest changes without the need of waiting a few seconds
- Plain app names were removed from code and moved to the langauge files (allows also translations of the app name).
- Many improvements and fixes

The complete list of changes can be found in the attached ChangesLog

<hr /><em>This discussion was created from the release <a href='https://github.com/numere-org/NumeRe/releases/tag/v1.1.5.2305'>v1.1.5 "Cherenkov"</a>.</em>

**Comment by numeredev** [comment]
Thanks go to the whole team!

If you experience issues, please get in touch. If you have further ideas or requests, feel free to open an issue here on GitHub.

**Comment by idealkindom** [comment]
I tried on PC witb win 10 mounted,on neither the help menu works.

Another thins is that as a beginning user I felt the example types are too few. Especially for tbe Graphical interface ,i.e., NumeRe Window Layout, I tried with failure, because I can not find a few examples and detailed explainations.

**Reply by numeredev** [reply]
Hello @idealkindom, thanks for your message.

Could you please attach some screenshots illustrating the issue with the help menu, please? We might be able to help you out there, although it's the first time, that we hear something like this.

Window Layouts are a "quite" new feature, but you're right that there are very few examples. Feel free get in touch with specific requirements, what you want to achieve with Window Layouts. The next version will offer a built-in generator for creating usable layouts, which you'll be able to adapt to your needs.

Cheers!

**Reply by idealkindom** [reply]
![Uploading 2023-11-16 18_43_43-Greenshot.png…]()

**Comment by idealkindom** [comment]
At 2023-11-15 21:47:02, "Erik Hänel" ***@***.***> wrote:

Hello @idealkindom, thanks for your message.

Could you please attach some screenshots illustrating the issue with the help menu, please? We might be able to help you out there, although it's the first time, that we hear something like this.

Window Layouts are a "quite" new feature, but you're right that there are very few examples. Feel free get in touch with specific requirements, what you want to achieve with Window Layouts. The next version will offer a built-in generator for creating usable layouts, which you'll be able to adapt to your needs.

Cheers!

**Reply by numeredev** [reply]
Hello @idealkindom,

if you tried to attach something via an email attachment, then GitHub might have refused to accept that. I'm afraid, but you need to add that here on GitHub as a comment directly.

Cheers!

**Reply by idealkindom** [reply]
<img width="960" alt="2023-11-16 18_43_43-Greenshot" src="https://github.com/numere-org/NumeRe/assets/18344868/9bab7ba9-734d-4823-a392-75480fe29e51">

**Reply by numeredev** [reply]
Hello @idealkindom,

thanks for the screenshot. That's strange behavior indeed. It seems that you are on a chinese or japanese system. The language files seem to be available and the help files index was also created, so they should actually be readable. Can you answer the following questions and perform the tasks, please?

- Is true that you tried to open the documentation browser by clicking on "Help" or pressing [F1]?
- Where is all that "light green" coming from? Is that a system setting of yours?
- Did you try to select any other entry in the index list on the left of the documentation browser? (If not, please do that for e.g. `clc`)
- Can you type into the terminal `edit <>/docs/main.nhlp` and check, whether the editor opens up an XML-like file, which doesn't look corrupted (usual text, no weird characters, ...)?
- Can you type in `findfile("<>/icons/folder.png")` and tell us, whether that returned `1`?
- Can you type in `help numere` or `help clc` and tell us, whether that changed something?
- Can you type in `#<>` and send us a screenshot of that? Should typically be s.th. like `"C:/Software/NumeRe"`

We hope that we can figure out, what's the problem on your side (hopefully it is only a missing file). Thank your for your collaboration.

Cheers!

**Reply by idealkindom** [reply]
Yes,it's on a Chinese  system. Answers to the questions and results of following the tasks:

Yes,I tried to open the documentation browser by clicking on "Help" or pressing [F1].

Yes,the "light green" comes a system setting of mine.

I tried to select entry in the index list on the left of the documentation browser?,but it doesn't work. When I hang the mouse on the commond it pop up hints，when I right click to open in help it works.

I typed into the terminal edit <>/docs/main.nhlp  and checked, but  the editor doesn't open up an XML-like file，the response is like the screenshot.

I typed in findfile("<>/icons/folder.png"), it returned 1 as the screenshot shows.

I typed in help numere and bugs the same like before.When I typed in help clc the help works.

Now I know the help could work but not in a most intuitive way.Thks a lot.
<img width="474" alt="2023-11-17 22_50_49-" src="https://github.com/numere-org/NumeRe/assets/18344868/83a6e7e6-bff3-4ef8-9b77-75ccab028972">
<img width="320" alt="2023-11-17 22_52_52-NumeRe_ Start here - NumeRe_ Framework for Numerical Calculations (v 1 1 5 _Cher" src="https://github.com/numere-org/NumeRe/assets/18344868/25b61b37-f92d-48dc-a0c5-8020e076ee98">
<img width="541" alt="2023-11-17 22_53_48-" src="https://github.com/numere-org/NumeRe/assets/18344868/4e421367-7082-41b3-afe2-725ccd031b2e">

**Reply by numeredev** [reply]
Hello @idealkindom,

thanks for the answers, regarding the `edit` command: you accidentally lost a whitespace there. Be sure to type `edit <>/docs/main.nhlp` with the whitespace between `edit` and `<>`.

So:
`help numere` works not.
`help clc` works.

It appears, that `main.nhlp` is probably corrupted (that's the file shown, if you click on "Help"). Either it is missing (which I do not assume, because the documentation browser should not open in such a case) or there's an error in this file. Have a look in the mentioned `<>/docs/main.nhlp` file. After the licence remark in the first lines it should look like so:
![grafik](https://github.com/numere-org/NumeRe/assets/87755490/45d3370c-565f-4b81-967b-c6c1cadb42b6)
I could replicate the error message by fiddling with the keywords. I hope that's your issue. If it is: it's probably best to download a complete new file from here (that's the revision built into your release): https://github.com/numere-org/Translation/tree/084e011b57c0fc245555881f52919fb2f9d62bcd/en-GB/docs

**Remark:** In case the file is missing, then the terminal will show you an error message that the file was not readable or could not be found. You can find its (intended) storage location, if you type in `edit <>/docs`. That should open up the corresponding folder. Download a new file from the repository link above and put it there.

**Reply by numeredev** [reply]
Another thing: you probably want to switch the background color of the editor to white or very-light gray. You can do that in the settings dialog at "Style" and setting the background color to 254,254,254 because pure white is probably ignored.

**Reply by idealkindom** [reply]
Hi, I'm trying to translate the files in the directory of docs and others, which starting from main.nhlp, but I found that it dose not work when I translated the text part into Chinese. I once thought that the syntax is the same to that  of html, for which all that I have to do is obeying the syntax rules and only change the text part.
[main.zip](https://github.com/numere-org/NumeRe/files/13759222/main.zip)

**Reply by numeredev** [reply]
Thanks for the start of the translation. We found that making Chinese work, that will require a set of internal changes. However, with your file we're now able to actually test, whether the changes will work. Making the mentioned changes, will take some time, so you'll have to be patient. Expect it not before mid of 2024. Here's the issue, if you want to keep track about that: https://github.com/numere-org/NumeRe/issues/172

**Reply by numeredev** [reply]
> Hi, I'm trying to translate the files in the directory of docs and others, which starting from main.nhlp, but I found that it dose not work when I translated the text part into Chinese. I once thought that the syntax is the same to that of html, for which all that I have to do is obeying the syntax rules and only change the text part. [main.zip](https://github.com/numere-org/NumeRe/files/13759222/main.zip)

With the current changes, we were able to load the documentation file you provided. One slight problem is the missing word wrap for CJK languages. Might only be fixable by introducing some "artificial" whitespaces between some Hanji so that the word wrapping actually has some wrapping candidates. Besides that, everything seem to look fine:
<img width="1040" height="793" alt="grafik" src="https://github.com/user-attachments/assets/9663a221-96d5-4e51-9f09-36337f262565" />

**Reply by idealkindom** [reply]
Oh, it looks quite good.Keep on nice job.

**Reply by numeredev** [reply]
With the new release candidate published, we may think about how to provide user-supplied languages. As a first thought, we probably won't directly include them into the installers, simply because they might not be be updated at release day. And bundling outdated language files will result in bad UX.

Therefore, keeping them in a separate `user` folder and requiring that users must download the community localizations themselves, is for the near future the only possible way to go.

Regarding the update process: I think, it is logical to create a diff file from the official documentation and translate only the new/changed strings. That is probably the easiest way to keep things updated. As a consequence, that requires that we tag our released language files as well. We'll need to document that process in our dedicated repository.

If you like to jump in for translating to Chinese, I recommend that you fork https://github.com/numere-org/Translation and create a new localization folder in a separate branch. I'd also recommend that you start translating the *.nlng files, which do contain the GUI language strings and there are only a few of them (notably, `numere.nlng`, `error.nlng` and `main.nlng`, the others are just template files). Refer to `help language` for further information. For testing, copy your edits to the folder at `edit <>/user/lang` and restart NumeRe.

**Comment by idealkindom** [comment]
Thanks a lot. I like NumeRe a lot at first. If it's possible, I would volunteer to translate the GUI to Chinese.However, I am not a programmer, I only made small short program to do engineering calculations with GNU Octave and Maxima.

在 2023-11-17 23:39:22，"Erik Hänel" ***@***.***> 写道：


Hello @idealkindom,

thanks for the answers, regarding the edit command: you accidentally lost a whitespace there. Be sure to type edit <>/docs/main.nhlp with the whitespace between edit and <>.

So:
help numere works not.
help clc works.

It appears, that main.nhlp is probably corrupted (that's the file shown, if you click on "Help"). Either it is missing (which I do not assume, because the documentation browser should not open in such a case) or there's an error in this file. Have a look in the mentioned <>/docs/main.nhlp file. After the licence remark in the first lines it should look like so:

I could replicate the error message by fiddling with the keywords. I hope that's your issue. If it is: it's probably best to download a complete new file from here (that's the revision built into your release): https://github.com/numere-org/Translation/tree/084e011b57c0fc245555881f52919fb2f9d62bcd/en-GB/docs

Remark: In case the file is missing, then the terminal will show you an error message that the file was not readable or could not be found. You can find its (intended) storage location, if you type in edit <>/docs. That should open up the corresponding folder. Download a new file from the repository link above and put it there.

**Reply by numeredev** [reply]
You can try to translate it file by file: the `<>/docs` as well the `<>/lang` folder each contain a `user` folder. You can start translating one file, put it in the `user` folder and see, whether that's accepted. You probably need to restart NumeRe after modifying these files. It might be the case that it isn't working. That may be the case due to the internal code page used. You can nevertheless send us one of the translated files, so that we can check, whether it is working in the current development version.

**Comment by idealkindom** [comment]
Thanks a lot, light green is backgroud color that I like the most.

在 2023-11-17 23:43:19，"Erik Hänel" ***@***.***> 写道：


Another thing: you probably want to switch the background color of the editor to white or very-light gray. You can do that in the settings dialog at "Style" and setting the background color to 254,254,254 because pure white is probably ignored.

**Reply by numeredev** [reply]
Then you probably want to switch some other syntax colors, which will probably be hard to read with that green background 😉

**Comment by idealkindom** [comment]
BTW, I could not understand why this happy on both of my 2 Win 10 PC. Maybe it's because of the   system lanuage?

在 2023-11-17 23:39:22，"Erik Hänel" ***@***.***> 写道：


Hello @idealkindom,

thanks for the answers, regarding the edit command: you accidentally lost a whitespace there. Be sure to type edit <>/docs/main.nhlp with the whitespace between edit and <>.

So:
help numere works not.
help clc works.

It appears, that main.nhlp is probably corrupted (that's the file shown, if you click on "Help"). Either it is missing (which I do not assume, because the documentation browser should not open in such a case) or there's an error in this file. Have a look in the mentioned <>/docs/main.nhlp file. After the licence remark in the first lines it should look like so:

I could replicate the error message by fiddling with the keywords. I hope that's your issue. If it is: it's probably best to download a complete new file from here (that's the revision built into your release): https://github.com/numere-org/Translation/tree/084e011b57c0fc245555881f52919fb2f9d62bcd/en-GB/docs

Remark: In case the file is missing, then the terminal will show you an error message that the file was not readable or could not be found. You can find its (intended) storage location, if you type in edit <>/docs. That should open up the corresponding folder. Download a new file from the repository link above and put it there.

**Reply by numeredev** [reply]
I do not know, was the problem that the file is corrupted?

It contains "Microsoft Windows(R)" within its contents, probably the system cannot handle the "registered" symbol.

**Comment by idealkindom** [comment]
OK, I will pay attentions to it.

在 2023-11-17 23:55:05，"Erik Hänel" ***@***.***> 写道：


Then you probably want to switch some other syntax colors, which will probably be hard to read with that green background 😉

**Comment by idealkindom** [comment]
I guess so, because after installation the shortcut name for NumeRe include some strange character after"NumeRe-Framework f"

在 2023-11-17 23:58:05，"Erik Hänel" ***@***.***> 写道：


I do not know, was the problem that the file is corrupted?

It contains "Microsoft Windows(R)" within its contents, probably the system cannot handle the "registered" symbol.

**Reply by numeredev** [reply]
Ahh, that is probably a "ü" for "für", which is German for "for" (sounds like recursion). Have to correct that in the installer. 

Can you check if the error persists, if you remove the "registered" symbol?

**Comment by idealkindom** [comment]
Al right. I will try later.

在 2023-11-18 00:01:37，"Erik Hänel" ***@***.***> 写道：


You can try to translate it file by file: the <>/docs as well the <>/lang folder each contain a user folder. You can start translating one file, put it in the user folder and see, whether that's accepted. You probably need to restart NumeRe after modifying these files. It might be the case that it isn't working. That may be the case due to the internal code page used. You can nevertheless send us one of the translated files, so that we can check, whether it is working in the current development version.

**Comment by idealkindom** [comment]
This time I correctly typed edit <>/docs/main.nhlp and it appears narmal, the format looks like html. I deleted the registered mark and now the help works perfectly.

Thanks a lot.

在 2023-11-18 00:18:32，"Erik Hänel" ***@***.***> 写道：


Ahh, that is probably a "ü" for "für", which is German for "for" (sounds like recursion). Have to correct that in the installer.

Can you check if the error persists, if you remove the "registered" symbol?

**Reply by numeredev** [reply]
Thank you for checking. We'll correct that in development and installer.

**Reply by numeredev** [reply]
[cheatsheet.pdf](https://github.com/numere-org/NumeRe/files/13414380/cheatsheet.pdf)
Hi @idealkindom,
here's a cheat sheet, which we created in the past (but forgot to upload to somewhere reasonable 😁 ).

**Comment by idealkindom** [comment]
Thanks a lot.

在 2023-11-20 20:26:36，"Erik Hänel" ***@***.***> 写道：


cheatsheet.pdf
Hi @idealkindom,
here's a cheat sheet, which we created in the past (but forgot to upload to somewhere reasonable 😁 ).
